### PR TITLE
Add FusedLinearCrossEntropy

### PR DESCRIPTION
**Pull request by FindHao**
As discussed in https://github.com/pytorch/pytorch/issues/136168, I'm going to migrate implementations of operator benchmarking. This PR adds different implementations for FusedLinearCrossEntropy as a starting example. 

Execution command:
```
python run_benchmark.py triton --op FusedLinearCrossEntropy
```
Example output:
```
x_val    LMHeadCE-latency    LigerLMHeadCE-latency    inductor_fused_linear_cross_entropy-latency
-------  ------------------  -----------------------  ---------------------------------------------
      0             98.0041                  389.87                                         95.0412
      1            196.12                    652.619                                       193.219
      2            417.242                  1248.75                                        416.725
      3            824.906                  2356.25                                        809.56
```

### CODE REVIEW

**Review by xuzhao9 — APPROVED** [review]

File: torchbenchmark/operators/FusedLinearCrossEntropy/operator.py [hunk 1]
```diff
@@ -0,0 +1,98 @@
+import argparse
+from typing import List, Optional, Generator, Callable
+import torch
+from torchbenchmark.util.triton_op import (
+    BenchmarkOperator,
+    register_benchmark,
+)
+from liger_kernel.transformers.fused_linear_cross_entropy import LigerFusedLinearCrossEntropyLoss
```
Comment (xuzhao9) [inline torchbenchmark/operators/FusedLinearCrossEntropy/operator.py:8]: We probably also need to setup how to install linger_kernel (e.g., adding it as a submodule, or install it in https://github.com/pytorch/benchmark/blob/main/userbenchmark/triton/install.py)

**Comment by xuzhao9** [comment]
Is liger_kernel only available in OSS but not fbcode? If so, we probably need to manually bypass the internal CI.
By default, the internal CI will load and test all operators, but linger_kernel is not available in fbcode.

File: torchbenchmark/operators/FusedLinearCrossEntropy/operator.py (hunk 1 above)
Comment (FindHao) [reply torchbenchmark/operators/FusedLinearCrossEntropy/operator.py:8]: add the installation for liger-kernel in 
 https://github.com/pytorch/benchmark/pull/2485/commits/9580b0f964b29f0599bbd872840a4540370e7234

Comment (xuzhao9) [reply torchbenchmark/operators/FusedLinearCrossEntropy/operator.py:8]: Bumping transformer version in https://github.com/pytorch/benchmark/pull/2488

**Review by eellison — APPROVED** [review]

File: torchbenchmark/operators/FusedLinearCrossEntropy/operator.py (hunk 1 above)
Comment (xuzhao9) [reply torchbenchmark/operators/FusedLinearCrossEntropy/operator.py]: The transformer version has been updated, can you help rebase this PR on the trunk?

Comment (FindHao) [reply torchbenchmark/operators/FusedLinearCrossEntropy/operator.py]: done

**Pull request closed without merging**